lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2346,22 +2346,37 @@ SubExprInfo ExprEmitter::visitSV(XMRRefOp op) {
   if (hasSVAttributes(op))
     emitError(op, "SV attributes emission is unimplemented for the op");
 
-  auto globalRef =
-      cast<hw::GlobalRefOp>(state.symbolCache.getDefinition(op.getRefAttr()));
-  auto namepath = globalRef.getNamepathAttr().getValue();
-  auto *module = state.symbolCache.getDefinition(
-      cast<InnerRefAttr>(namepath.front()).getModule());
-  ps << getSymOpName(module);
-  for (auto sym : namepath) {
-    ps << ".";
-    auto innerRef = cast<InnerRefAttr>(sym);
+  auto refAttr = op.getRefAttr();
+
+  // The XMR is pointing at an InnerRefAttr.
+  if (auto innerRef = dyn_cast<InnerRefAttr>(refAttr)) {
     auto ref = state.symbolCache.getInnerDefinition(innerRef.getModule(),
                                                     innerRef.getName());
-    if (ref.hasPort()) {
+    ps << getSymOpName(state.symbolCache.getDefinition(innerRef.getModule()));
```

Comment:
Nit: getSymOpName and getPortVerilogName can be wrapped with `PPExtString` to avoid copying them to temporary string storage while printing.

Nevermind, this is missed in a few places already (whoops) and if that's the contract might be better to just have them return PPExtString's directly or otherwise try to make it not something folks have to worry about, this isn't... awesome.  I'll tackle this separately, but leaving this here as a small flag to consider this when hacking on ExportVerilog please :+1: .

Benchmarking the PrettyPrinter getting this right in most places helped a bit re:perf/memory (we end up doing less copying than pre-pretty-printer this way in some cases!) but the this is probably not on the hot path :smile: .

Reply:
Good suggestion. I switched this over to use `PPExtString` in the force push.